posts/2024-02-05-crates-io-status-codes.md

```diff
@@ -0,0 +1,35 @@
+---
+layout: post
+title: "crates.io: API status code changes"
+author: Tobias Bieniek
+team: the crates.io team <https://www.rust-lang.org/governance/teams/crates-io>
+---
+
+When cargo and crates.io were first implemented, [HTTP response status codes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status) were ignored. cargo assumed that crates.io would always respond with a "[200 OK](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200)" and any error messages are detected from the response JSON body. When crates.io responded with a different status code then cargo was showing the raw JSON body instead of a regular error message:
```

Comment:
I think we should captialize Cargo:

```suggestion
When Cargo and crates.io were first implemented, [HTTP response status codes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status) were ignored. Cargo assumed that crates.io would always respond with a "[200 OK](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200)" and any error messages are detected from the response JSON body. When crates.io responded with a different status code then Cargo was showing the raw JSON body instead of a regular error message:
```

Reply:
@rust-lang/cargo any thoughts? I tend to use the casing of the CLI tool (aka. lowercase) but if the official spelling is uppercase let me know :)